cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -59,23 +61,72 @@ const (
 	HintInhibitedForPreDownload Hint = "pre-download"
 )
 
+const hintFilePostfix = "lock"
+
 // HintFile returns the full path of the run inhibition lock file for the given snap.
 func HintFile(snapName string) string {
-	return filepath.Join(InhibitDir, snapName+".lock")
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hintFilePostfix))
+}
+
+func InhibitInfoFile(snapName string, hint Hint) string {
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hint))
 }
 
 func openHintFileLock(snapName string) (*osutil.FileLock, error) {
 	return osutil.NewFileLockWithMode(HintFile(snapName), 0644)
 }
 
+type InhibitInfo struct {
+	Revision snap.Revision `json:"revision"`
+}
+
+func validateInhibitInfo(info InhibitInfo) error {
+	if info.Revision.Unset() {
+		return fmt.Errorf("snap revision cannot be unset")
+	}
+	return nil
+}
+
+func validateHint(hint Hint) error {
+	if len(hint) == 0 {
+		return fmt.Errorf("lock hint cannot be empty")
+	}
+	if string(hint) == hintFilePostfix {
+		return fmt.Errorf("hint cannot have value %q", hintFilePostfix)
+	}
+	return nil
+}
+
+func removeInhibitInfoFiles(snapName string) error {
+	infoGlob := filepath.Join(InhibitDir, snapName+".*")
+	// There should be one file only, but just in case
+	files, err := filepath.Glob(infoGlob)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		// Don't remove hint
+		if filepath.Base(f) == fmt.Sprintf("%s.%s", snapName, hintFilePostfix) {
+			continue
+		}
+		if err := os.Remove(f); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // LockWithHint sets a persistent "snap run" inhibition lock, for the given snap, with a given hint.
 //
 // The hint cannot be empty. It should be one of the Hint constants defined in
 // this package. With the hint in place "snap run" will not allow the snap to
 // start and will block, presenting a user interface if possible.
```

Comment:
this doc comment should talk about info as well now

Reply:
updated, thanks!